kaitai/stream.go

```diff
@@ -63,19 +63,17 @@ func (k *Stream) Size() (int64, error) {
 	if err != nil {
 		return 0, err
 	}
+	// Deferred Seek back to the current position
+	defer k.Seek(curPos, io.SeekStart)
```

Comment:
The tool https://github.com/golangci/golangci-lint noticed this problem:

```console
$ ./golangci-lint-1.35.2-windows-amd64/golangci-lint run kaitai
kaitai\stream.go:67:14: Error return value of `k.Seek` is not checked (errcheck)
        defer k.Seek(curPos, io.SeekStart)
                    ^
```

This might potentially lead to a problem that **this call** fails, but because its error return value is ignored, `k.Size()` will happily return `error == nil` (despite the fact that the stream pointer failed to return to the original position for some reason and we're clearly in a situation of failure).

I think the code should be like this:

```diff
-func (k *Stream) Size() (int64, error) {
+func (k *Stream) Size() (int64, err error) {
```

```suggestion
	defer func() {
		if _, serr := k.Seek(curPos, io.SeekStart); serr != nil {
			err = serr
		}
	}()
```

See https://yourbasic.org/golang/defer/#use-func-to-return-a-value and https://blog.learngoprogramming.com/5-gotchas-of-defer-in-go-golang-part-iii-36a1ab3d6ef1#afb5.

Could you please adjust the tests to check this case as well?

Perhaps we can make the `failingReader` more generic by restoring its normal behavior (i.e. `io.SeekEnd` behaving as normal, not setting `pos` to `-1`) and instead passing a param in it (e.g. `numSeeksLimit`) that limits the number of succeeding `Seek()` calls. Then we can just go iterate 0..(total number of `Seek()`s the tested function does) and check for each limit that the `Size()` function always returns an error and restores the stream position to its original state.

Reply:
I've added two test cases:
1. fails to get the initial pos
2. the deferred seek fails